src/v/iceberg/unresolved_partition_spec.h

```diff
@@ -21,13 +21,21 @@ struct unresolved_partition_spec {
         transform transform;
         ss::sstring name;
 
+        friend bool operator==(const field&, const field&) = default;
         friend std::ostream& operator<<(std::ostream&, const field&);
     };
 
     chunked_vector<field> fields;
 
+    friend bool operator==(
+      const unresolved_partition_spec&, const unresolved_partition_spec&)
+      = default;
+
     friend std::ostream&
     operator<<(std::ostream&, const unresolved_partition_spec&);
+
+    static std::optional<unresolved_partition_spec>
+    parse(const std::string_view&);
```

Comment:
Just wondering if these semantics are defined anywhere in the Iceberg spec, or if this is something else? I suppose it's somewhat a subset of SQL DDL?

Reply:
Also if it isn't part of the Iceberg spec, may be worth moving into v/datalake to avoid confusion

Reply:
No, this is not part of the spec, it mainly follows spark-sql DDL for partition expressions. I'll move it out, though this will require introducing yet another library :)